src/main/java/Marites.java

```diff
@@ -0,0 +1,198 @@
+     * @param feedback The feedback string to execute.
+     */
+    private static void showFeedback(String feedback) {
+        printSeparator();
+        System.out.print(feedback);
+        printSeparator();
+    }
+
+    /**
+     * Get the type of command issued.
+     * @param userInput The user's input
+     * @return The command type.
+     */
+    private static String getCommandType(String userInput) {
+        String[] tokens = userInput.split("\\s");
+        return tokens[0];
+    }
+
+    /**
+     * Parses a user's command, and executes it.
+     * @param userInput A line of the user's input.
+     * @return Output that must be passed back to the user.
+     */
+    private static String processUserCommand(String userInput) {
+        switch (getCommandType(userInput)) {
+        case COMMAND_EXIT:
+            return executeExit();
+        case COMMAND_LIST:
+            return executeListTasks();
+        case COMMAND_MARK:
+            return executeSetTaskStatus(userInput, true);
+        case COMMAND_UNMARK:
+            return executeSetTaskStatus(userInput, false);
+        case COMMAND_ADD_TODO:
+        case COMMAND_ADD_DEADLINE:
+        case COMMAND_ADD_EVENT:
+            return executeAddTask(userInput);
+        default:
+            return UNKNOWN_COMMAND_MESSAGE;
+        }
+    }
+
+    /*
+        The following methods take in a String,
+            denoting the full line of user input.
+        They return a String, denoting the feedback after executing the command.
+     */
+
+    /**
+     * Executes a list command.
+     * @return The list of tasks.
+     */
+    private static String executeListTasks() {
+        StringBuilder taskList = new StringBuilder();
+        for (int i = 1; i <= tasks.size(); ++i) {
+            String task = String.format(LIST_TASK_ITEM_FORMAT_STRING, i, tasks.get(i-1));
+            taskList.append(task);
+        }
+        return taskList.toString();
+    }
+
+    /**
+     * Parses an add task command given by the user.
+     * @param userInput The user's input
+     * @return A Task object representing the task
+     */
+    private static Task parseAddTask(String userInput) {
+        String[] tokens = userInput.split("\\s", 2);
+        String[] parametersSplit;
+        String commandType = tokens[0], commandParameters = tokens[1];
+        switch (commandType) {
+        case COMMAND_ADD_TODO:
+            return new Todo(commandParameters.strip());
+        case COMMAND_ADD_DEADLINE:
+            parametersSplit = commandParameters.split(COMMAND_ADD_DEADLINE_TAG);
+            return new Deadline(parametersSplit[0].strip(), parametersSplit[1].strip());
+        case COMMAND_ADD_EVENT:
+            parametersSplit = commandParameters.split(COMMAND_ADD_EVENT_TAG);
+            return new Event(parametersSplit[0].strip(), parametersSplit[1].strip());
+        default:
+            return new Task("");
+        }
+    }
```

Comment:
It might help to merge this function with `processUserCommand` since you already parse it enough there to know what kind of command you are getting.

Reply:
I agree with the concern about repetition of logic, but I'm concerned the method will be too long. I've made `getCommandType` also return the rest of the command (and renamed it appropriately), so that the succeeding methods will only have to further parse the rest of the command.